### PR TITLE
Simplify code generator dependencies

### PR DESCRIPTION
**Pull request by liggitt**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Since code-generator is at the root of all k8s deps, simplifying dependencies there helps all other components. We had a large dependency tree used only for import ordering in a single method by code-gen. Rewrote that method and dropped many dependencies for most of our published repos.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area code-organization
/cc @dims
for review
 
/cc @sttts
who added the import sorting

### CODE REVIEW

**Comment by dims** [comment]
whoa! thanks @liggitt

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go [hunk 1]
```diff
@@ -374,38 +371,73 @@ func deps(c *generator.Context, pkgs []*protobufPackage) map[string][]string {
...
 		}
 	}
 
-	ret := []string{}
-	sorted, err := topo.Sort(g)
-	if err != nil {
-		return nil, err
+	// find initial nodes without any dependencies
+	sorted := findAndRemoveNodesWithoutDependencies(remainingNodes, graph)
+	for i := 0; i < len(sorted); i++ {
```
Comment (dims) [inline staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go]: interesting for loop :) we use `len` so we iterate over the newly appended nodes.

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go [hunk 2]
```diff
@@ -374,38 +371,73 @@ func deps(c *generator.Context, pkgs []*protobufPackage) map[string][]string {
...
+}
+
+// edge describes a from->to relationship in a graph
+type edge struct {
+	from string
+	to   string
+}
+
+// findAndRemoveNodesWithoutDependencies finds nodes in the given set which are not pointed to by any edges in the graph,
+// removes them from the set of nodes, and returns them in sorted order
```
Comment (dims) [inline staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go]: quick q : does sorting here matter?

**Comment by dims** [comment]
/assign

one observation and quick question inline @liggitt

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go (hunk 2 above)
Comment (liggitt) [reply staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go]: I sort lexically at the end, before returning, to make the output deterministic. That's actually an improvement over the preexisting code, which was nondeterministic. The order doesn't actually matter for our purposes as long as it is topologically sorted, so we process dependencies before the things that depend on them

File: staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf/cmd.go]: Yes

**Comment by dims** [comment]
/approve
/lgtm

thanks @liggitt

**Pull request merged**